gslib/tests/test_rsync.py

```diff
@@ -2781,7 +2781,18 @@ def _Check1():
           set(['/a', '/b', '/c', '/data1/a.txt', '/data1/ok', '/data2/b.txt']))
       self.assertEquals(listing2, set(['/a', '/b', '/c', '/data1/ok']))
 
+    # Use @Retry as hedge against bucket listing eventual consistency.
+    @Retry(AssertionError, tries=3, timeout_secs=1)
+    def _Check2():
+      """Tests that a regex with a pipe works as expected."""
+      bucket_uri = self.CreateBucket()
+      regex = '.*\\.txt$|ok$'
```

Comment:
Shouldn't it be `'.*\\.txt$|.*ok$'`? I'm very confused why it works currently, any idea?

 Maybe also add a test for `^`, because that's the main reason we had to go back to the tuple solution.

Reply:
It works because there's no `^`...! `^ok$` wouldn't work, but here I'm just saying exclude anything that ends in "ok". But yes, we should include a test for `^`, but I would argue one that doesn't follow with a `.*` so we can confirm it's not checking the base URL.